compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java

```diff
@@ -3827,6 +3829,22 @@ private boolean evaluateRawTemplateExprs(List<? extends BLangExpression> exprs,
         return errored;
     }
 
+    private BType getResolvedIntersectionType(BType type) {
+        return type.tag != TypeTags.INTERSECTION ? type : ((BIntersectionType) type).effectiveType;
+    }
+
+    private boolean containsAnyType(BType type) {
+        if (type == symTable.anyType) {
+            return true;
+        }
+
+        if (type.tag == TypeTags.UNION) {
```

Comment:
What if the union has `any` and one or more compatible object types? In that case it should be an ambiguous type error right?

Reply:
Shouldn't a union of `any` and any other type (except error) simplify to just `any` (currently it doesn't it seems)? In which case it should just be considered a `RawTemplate`

Reply:
Shouldn't we handle this scenario for the time-being though?

Reply:
This behaviour is correct for the scenario right? Considering that `any|Foo|Bar` (where `Foo` and `Bar` are raw template types) is the same as `any`